lib/collection/src/operations/types.rs

```diff
@@ -328,6 +328,27 @@ pub struct ShardTransferInfo {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[anonymize(false)]
     pub comment: Option<String>,
+
+    /// Internal status of the shard transfer, do not expose.
+    #[serde(skip)]
+    #[schemars(skip)]
+    #[anonymize(false)]
+    pub status: ShardTransferStatus,
+}
+
+#[derive(Debug, Clone, Anonymize)]
+pub struct ShardTransferStatus {
+    #[anonymize(false)]
+    pub running: Option<bool>,
+
+    #[anonymize(false)]
+    pub failed: Option<bool>,
+
+    #[anonymize(false)]
+    pub points_transferred: Option<usize>,
+
+    #[anonymize(false)]
+    pub points_total: Option<usize>,
 }
```

Comment:
Why not just reuse `TransferTaskStatus`? And I think optional struct is makes more sense than struct with optional fields.

```suggestion
    pub status: Option<ShardTransferStatus>,
}
```

Reply:
Addressed in [8ad0c0f](https://github.com/qdrant/qdrant/pull/7714/commits/8ad0c0f073351ec874d04318998e990a0a04e6a0)